neurodsp/sim/multi.py

```diff
@@ -0,0 +1,166 @@
+    """Generator to yield simulated signals from a parameter sampler.
+
+    Parameters
+    ----------
+    sim_func : callable
+        Function to create the simulated time series.
+    sim_params : iterable
+        The parameters for the simulated signal, passed into `sim_func`.
+    return_sim_params : bool, optional, default: False
+        Whether to yield the simulation parameters as well as the simulated time series.
+    n_sims : int, optional
+        Number of simulations to set as the max.
+        If None, creates an infinite generator.
+
+    Yields
+    ------
+    sig : 1d array
+        Simulated time series.
+    sample_params : dict
+        Simulation parameters for the yielded time series.
+    """
+
+    if len(sim_params) and n_sims and n_sims > len(sim_params):
+        msg = 'Cannot simulate the requested number of sims with the given parameters.'
+        raise ValueError(msg)
+
+    for ind, sample_params in zip(counter(n_sims), sim_params):
+
+        if return_sim_params:
+            yield sim_func(**sample_params), sample_params
+        else:
+            yield sim_func(**sample_params)
+
+        if n_sims and ind >= n_sims:
+            break
+
+
+def sim_multiple(sim_func, sim_params, n_sims):
+    """Simulate multiple samples of a specified simulation.
+
+    Parameters
+    ----------
+    sim_func : callable
+        Function to create the simulated time series.
+    sim_params : dict
+        The parameters for the simulated signal, passed into `sim_func`.
+    n_sims : int
+        Number of simulations to create.
+
+    Returns
+    -------
+    sigs : 2d array
+        Simulations, as [n_sims, sig length].
+    """
+
+    sigs = np.zeros([n_sims, sim_params['n_seconds'] * sim_params['fs']])
+    for ind, sig in enumerate(sig_yielder(sim_func, sim_params, n_sims)):
+        sigs[ind, :] = sig
+
+    return sigs
+
+
+def sim_across_values(sim_func, sim_params, n_sims, output='dict'):
+    """Simulate multiple signals across different parameter values.
+
+    Parameters
+    ----------
+    sim_func : callable
+        Function to create the simulated time series.
+    sim_params : iterable or list of dict
+        Simulation parameters for `sim_func`.
+    n_sims : int
+        Number of simulations to create per parameter definition.
+    output : {'dict', 'array'}
+        Organization of the output for the sims.
+        If 'dict', stored in a dictionary, organized by simulation parameter.
+        If 'array', all sims are organized into a 2D array.
+
+    Returns
+    -------
+    sims : dict of {float : array} or array
+        If dict, dictionary of simulated signals, where:
+            Each key is the simulation parameter value for the set of simulations.
+            Each value is the set of simulations for that value, as [n_sims, sig_length].
+        If array, is all signals collected together as [n_sims, sig_length].
+    """
```

Comment:
Having a float as a key feels odd. Maybe a list of [{param: float, sig: array}, ...]? 

Or a 1d array of float (params) and a 2d array (sigs), separately returned? And then replace `output` kwarg with `return_params=bool`. -> just saw this is done in `sim_sampler` so maybe that would work here?

Reply:
Yeh, so the more I thought about this, and trying to think about something like a structured dictionary of the kind you mentioned - at some point I realized that really that just becomes an object. So, having fallen down that rabbit hole, I've updated this PR to add a couple basic objects to manage groups of simulations (and related metadata, like parameters).  The sim multi functions now default to returning these objects (though it can be specified to return arrays, with no metadata). I think this is probably the best way to go here - everything is now updated